packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -144,6 +145,38 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
                 }
         )
     }
+
+    @VisibleForTesting
+    fun onDelete(flutterResult: Result, request: HashMap<String, Any>) {
+        try {
+            var modelName = request["modelName"] as String
+            var modelData = request["model"] as HashMap<String, Any>
+            var queryOptions: QueryOptions = QueryOptionsBuilder.fromSerializedMap(request)
+            val plugin = Amplify.DataStore.getPlugin("awsDataStorePlugin") as AWSDataStorePlugin
+
+            var instance = SerializedModel.builder()
+                    .serializedData(modelData)
+                    .id(modelData["id"] as String)
+                    .modelName(modelName)
+                    .build()
+
+            plugin.delete(
+                    instance,
+                    queryOptions.queryPredicate,
+                    Consumer {
+                        LOG.info("Deleted item: " + it.item().toString())
+                        handler.post { flutterResult.success(FlutterSerializedModel(it.item()).toMap()) }
+                    },
+                    Consumer {
+                        LOG.error("Deletion Failed: " + it)
+                        prepareError(flutterResult, it, FlutterDataStoreFailureMessage.AMPLIFY_DATASTORE_DELETE_FAILED.toString())
+                    }
+            )
+        } catch (e: Exception) {
+            prepareError(flutterResult, e, FlutterDataStoreFailureMessage.AMPLIFY_DATASTORE_DELETE_FAILED.toString())
```

Comment:
We should handle and throw malformed request errors as well.

Reply:
Done.